editor/editor/mode-switcher.js

```diff
@@ -0,0 +1,54 @@
+class ModeSwitcher extends wp.element.Component {
+	constructor() {
+		super( ...arguments );
+		this.toggle = this.toggle.bind( this );
+		this.state = {
+			opened: false
+		};
+	}
+
+	toggle( event ) {
+		event.preventDefault();
```

Comment:
>If you assign `type="button"` to the `button` element you can avoid the need for `preventDefault` (only needed because [default `type` of `button` is `"submit"`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type)).

I guess it makes no difference at the moment since the button lives outside the page form, but having not assigned a `type` to the button may come to bite us in the future.